### PR TITLE
Add ability to exclude Gerrit changes with no reviewers invited

### PR DESCRIPTION
**Pull request by danrodrig**
It's common to have for some time Gerrit changes without reviewers invited. For example, the CI failed and the change needs a new PS to be fixed. In these cases, these changes shouldn't be counted as reviews.
This change adds ability to skip changes with no reviewers and also allows to exclude users from the reviewers in the change. This can be used to not count as reviewers bot accounts used, for example, by the CI.

### CODE REVIEW

**Comment by amarza-rh** [comment]
Hi, please rebase from master and fix merge conflicts.

**Comment by pbortlov** [comment]
@danrodrig please ^

**Comment by danrodrig** [comment]
@amarza-rh @pbortlov rebased

File: reviewrot/gerritstack.py [hunk 1]
```diff
@@ -52,15 +51,15 @@ def request_reviews(self, host, repo_name, age=None,
...
-        if self.host_exists and repo_exists:
-            request_url = "{}/changes/?q=project:{}+status:open&" \
-                          "o=DETAILED_ACCOUNTS".format(self.url, repo_name)
-            log.debug('Looking for change requests for %s -> %s',
-                      self.url, repo_name)
-            review_response = self._call_api(url=request_url,
-                                             ssl_verify=ssl_verify)
-            reviews = self.format_response(review_response, age, show_last_comment)
-        return reviews
+        if not (self.host_exists and repo_exists):
```
Comment (amarza-rh) [inline reviewrot/gerritstack.py]: I would rephrase this as `if not self.host_exists or not repo_exists`, just so readers don't have to perform De Morgan's in their heads :)

File: test/gerrit_tests/test_gerrit.py [hunk 2]
```diff
@@ -388,6 +397,145 @@ def test_request_reviews_no_repos(self,
         mock_format_response.assert_not_called()
         self.assertEqual(None, response)
 
+    @patch(PATH + 'GerritService.format_response')
```
Comment (amarza-rh) [inline test/gerrit_tests/test_gerrit.py]: where does PATH come from?

File: test/gerrit_tests/test_gerrit.py [hunk 3]
```diff
@@ -388,6 +397,145 @@ def test_request_reviews_no_repos(self,
...
+        service = GerritService()
+
+        service.request_reviews(
+            'host', 'repo', reviewers_config=reviewers_config
+        )
+
+        assert mocked_requests.session.called
+        assert mocked_get_response.called
+        assert mocked_filter_invited.called
+        called_args, _ = mocked_filter_invited.call_args
```
Comment (amarza-rh) [inline test/gerrit_tests/test_gerrit.py]: can this section be refactored to `mocked_filter_invited.called_once_with(...)`?

File: test/gerrit_tests/test_gerrit.py (hunk 2 above)
Comment (danrodrig) [reply test/gerrit_tests/test_gerrit.py]: It's a global in test_gerrit.py. Existing tests were already using it.

File: reviewrot/gerritstack.py (hunk 1 above)
Comment (danrodrig) [reply reviewrot/gerritstack.py]: Done.

File: test/gerrit_tests/test_gerrit.py (hunk 3 above)
Comment (danrodrig) [reply test/gerrit_tests/test_gerrit.py]: Yep. I changed the asserts for consistency with the other tests.

**Comment by amarza-rh** [comment]
LGTM now
